### DISCUSSION TITLE
Include global folder via settings.json

### DISCUSSION THREAD

**Discussion by wolflu05** [James-Yu/LaTeX-Workshop#4199] [Q&A] [answered]
Hello,

I want to use this extension to write letters and therefore I need to include my signature as a picture in most of the documents. I could now copy the signature image next to all letter `.tex` files and include them, but this is not a good solution in my opinion. Accesing it by a predefined absolut path is also not the best idea, because if the environment changes from one pc to another pc the path to the signature may be different.

 Is there any option to include a directory or variable or ... in the `settings.json` where I can configure the path to the signature on pc level and in the LaTeX documents just use that path in the `\includegraphics[...]{path}`?

**Comment by jlelong** [comment]
Unfortunately, it is not possible for the latex compile to access anything from `settings.json`. 
If your question is a compiling issue, you can use the `TEXINPUTS` environment variable. See https://tex.stackexchange.com/questions/93712/definition-of-the-texinputs-variable

You could even specify it in the recipe using the `env` property, see https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-tools

**Reply by wolflu05** [reply]
Thank you for your quick reply. I also found out about `TEXINPUTS`, but I'm not sure how this could help me. Do you have a short example how I could achieve my goal with it?

**Reply by jlelong** [reply]
In your `.tex` file, you just write `\includegraphics[...]{mysignaturefile}`. To compile it, you run 

```
TEXINPUTS="the/fullpath/to/mysignaturefile:$TEXINPUTS" pdflatex myfile.tex
```

You can also have a look at https://www2.ph.ed.ac.uk/~wjh/tex/documents/environmental.pdf

**Reply by wolflu05** [reply, marked as answer]
Thank you very much for your quick help, I think that's exactly what I'm looking for. 

It took me a lot of playing around with different methods, because they do not worked correctly, but the following now works:

```json
  "latex-workshop.latex.tools": [
    {
      "name": "latexmk",
      "command": "latexmk",
      "args": [
        "-synctex=1",
        "-interaction=nonstopmode",
        "-file-line-error",
        "-pdf",
        "-outdir=%OUTDIR%",
        "%DOC%"
      ],
      "env": {
        "TEXINPUTS": "/path/to/folder/with/signarure/image//:",
      }
    }
  ]
```

Important here is that I use the folder and not the image and add `//:` to the end. As I read: 
- `//` means search in the folder recursively (even that the file is directly in the folder it doesn't work otherwise with just a single `/`).
- `:` at the end means append that default path here. (When using the `TEXINPUTS=/path/to/folder/:$TEXINPUTS` approach you'll get an error that a ENV references itself)
